src/lib/construct-line.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+function constructLines(line) {
+  const lines = [];
+  line.clusters_x.map((point, i) => {
+    if (i > 0) {
+      lines.push(
+        <line
+          key={`line${i-1}`}
+          x1={line.clusters_x[i - 1]} y1={line.clusters_y[i - 1]}
+          x2={line.clusters_x[i]} y2={line.clusters_y[i]}
+          stroke="#c33" strokeWidth="2"
+        />
+      );
+    }
+  });
+  return lines;
+}
+
+function constructPoints(line) {
+  const points = [];
+  line.clusters_x.map((point, i) => {
+    const x = line.clusters_x[i];
+    const y = line.clusters_y[i];
+    points.push(
+      <circle
+        key={`point${i}`}
+        cx={x} cy={y} r={5} fill="#c33"
+      />
+    );
+  });
+  return points;
+};
+
+function constructText(line) {
+  const sentences = [];
+  line.clusters_text.map((value) => {
+    value.map((word, i) => {
+      if (!sentences[i]) { sentences[i] = []; }
+      if (word.length) { sentences[i].push(word); }
```

Comment:
I'm not sure how a user would be able to pass in `null` here, and I'm trying to think of how this could potentially appear in `clusters_text`. As far as a string being `'word.length'`, I think that would be fine if that's what a user enters as a classification (if I understand this question correctly).